v2/internal/util/match/string_matcher.go

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package match
+
+import (
+	"fmt"
+)
+
+type Result struct {
+	Matches         bool
```

Comment:
Nitpick on naming.

```suggestion
	Matched         bool
```

(Since, by the time anyone reads the field, the value indicates whether something matched in the past.)